packages/rendermime-extension/src/index.ts

```diff
@@ -73,11 +73,12 @@ function activate(
             // Look for the an element with the hash id in the document.
             // This id is set automatically for headers tags when
             // we render markdown.
-            const element = widget.node.querySelector(id);
-            if (element) {
-              element.scrollIntoView();
-            }
-            return;
+            widget.context.session.ready.then(() => {
```

Comment:
I think waiting on the session is kind of confusing since not all documents have client sessions. For instance, markdown documents don't have active sessions by default. They do get a do-nothing session, so this does still work for them, but I think it's kind of unclear and possibly more error-prone.

I think we can just wait for `widget.context.ready`, which is closer to what is needed for the fix.

Reply:
Is there not a `widget.ready` for when it has finished rendering?  I'm afk

Reply:
Good point. The `IDocumentWidget` (which is returned by `docmanager.openOrReveal`) has a `revealed` promise on it. That promise is probably the best choice (I think it defaults to the same as  `context.ready`).